src/Processors/Formats/Impl/Parquet/ParquetBloomFilterCondition.cpp

```diff
@@ -173,6 +190,11 @@ std::vector<uint64_t> hash(const IColumn * data_column, const parquet::ColumnDes
         }
     }
 
+    if (hashes.size() != data_column->size())
```

Comment:
Wait, how does this not fail when `hashes.empty()`? How do tests pass?

Reply:
I missed this case, sorry. Fixed by returning optional vector. Also added a test that confirmed the `LOGICAL_ERROR` exception.